src/main/webapp/js/script.js

```diff
@@ -0,0 +1,61 @@
+window.addEventListener('DOMContentLoaded', (event) => {
+  const cards = document.getElementsByClassName('card');
+  initVote(cards);
+  for (const card of cards) {
+    const up = card.getElementsByClassName('upvote')[0];
+    const down = card.getElementsByClassName('downvote')[0];
+    up.addEventListener('click', function() {
+      renderVote(card, 'upvote');
+    });
+    down.addEventListener('click', function() {
+      renderVote(card, 'downvote');
+    });
+  }
+});
+
+function initVote(cards) {
```

Comment:
This shouldn't be necessary. Can't this be done on initial render? But fine, I'll okay it.

Reply:
In my first implementation of this color loading, I did all the rendering, color included in SearchServlet inside the card that is returned but you mentioned that server should avoid being responsible for anything UI styling related.

Reply:
I meant that the API (the json returned by GET/POST requests) should not contain specific UI styling.

It's a bit confusing, but the server-rendered template (the JSP template) is a client of your server. It receives the cards from the servlet, and figures out how to render them in HTML, even though technically this code is part of the server.

The reason I'm saying it's fine is because you would need to add a new field to Card.java, and have both JSP code and JS code to change the color, and that's more work than I'm willing to ask you to do.